acceptance/src/data.rs

```diff
@@ -7,6 +7,7 @@ use arrow_select::{concat::concat_batches, filter::filter_record_batch, take::ta
 
 use delta_kernel::{engine::arrow_data::ArrowEngineData, DeltaResult, Engine, Error, Table};
 use futures::{stream::TryStreamExt, StreamExt};
+use itertools::Itertools;
```

Comment:
is itertools for `try_collect()`? can we just `collect` into Result?

Reply:
`try_collect` is a thin wrapper around a result-returning `collect` call, but with the big advantage that it constrains types enough for type inference to work reliably. So where this usually won't work:
```rust
let foo: Vec<_> = iter.map(...).collect()?;
```
this usually _does_ work:
```rust
let foo: Vec<_> = iter.map(...).try_collect()?;
```